### ISSUE TITLE
Unmatched `(` and `)` in Strings stall REPL

### ISSUE THREAD

**Issue by andrewconner** [lihaoyi/Ammonite#283]
It seems that strings with `(` and `)` in them stall the REPL if they aren't matched. Examples:

```
@ "("
// Had to Ctrl+C to kill
@ ""
res0: String = ""
@ ")"
// Had to Ctrl+C to kill
@ "a"
res1: String = "a"
@ "a)"
// Had to Ctrl+C to kill
@ "(a)"
res2: String = "(a)"
@ "()"
res3: String = "()"
@ "())"
// Had to Ctrl+C to kill
```

Discovered because it stalls on this:
```
@ "".split("\\(")
// Had to kill with Ctrl+C
```

Ammonite Repl 0.5.0 (Scala 2.11.7 Java 1.8.0_25)

**Comment by andrewconner** [comment]
Actually I didn't properly minimize the bug. You can accomplish this much more easily. Updating issue.

**Comment by lihaoyi** [comment]
lol turns out that I had the dev-test-terminal's multiline filter included in the main REPL's set of filters

https://github.com/lihaoyi/Ammonite/blob/master/terminal/src/main/scala/ammonite/terminal/BasicFilters.scala#L14

This should be an easy fix

**Comment by lihaoyi** [comment]
Nice find @andrewconner

**Comment by andrewconner** [comment]
Verified, fixed in 0.5.1
